force-app/main/service/DonationHistoryService.cls

```diff
@@ -100,7 +100,24 @@ public with sharing class DonationHistoryService {
         } else {
             throw new UTIL_Permissions.InsufficientPermissionException(Label.commonInsufficientPermissions);
         }
-        
+    }
+
+    public String getRequiredPaymentMethodForOpportunities(List<npe01__OppPayment__c> paymentListForOpportunity){
+        String paymentMethodToReturn = '';
+		if(paymentListForOpportunity.Size() > 1){
+            for(npe01__OppPayment__c p : paymentListForOpportunity){
+                if(p.npe01__Paid__c){
+                    paymentMethodToReturn= p.npe01__Payment_Method__c;
+                    break;
+                }
+           	}
+            if(paymentMethodToReturn==''){
+                paymentMethodToReturn =  paymentListForOpportunity.get(0).npe01__Payment_Method__c;
+            }
+        }else{
+            paymentMethodToReturn =  paymentListForOpportunity.get(0).npe01__Payment_Method__c;
```

Comment:
I think this approach is better, your code has some potential problems(you ask if size > 1 but then you use the first element assuming is there, which is something you cannot warranty; you have multiple if and else and there is some indentation problems)

```
paymentMethodToReturn = ''
for (npe01__OppPayment__c p : paymentListForOpportunity) {
    if (paymentMethodToReturn == '') {
        paymentMethodToReturn = p.npe01__Payment_Method__c;
        continue;
    }
   if(p.npe01__Paid__c){
        paymentMethodToReturn = p.npe01__Payment_Method__c;
        break;
    }
}
return paymentMethodToReturn;
```